src/gateway/tasks/stats.ts

```diff
@@ -0,0 +1,219 @@
+    pageInfo {
+        hasNextPage
+      }
+      edges {
+        node {
+          id
+          owner {
+            address
+          }
+          tags {
+            name
+            value
+          }
+          fee { 
+            winston
+            ar
+          }
+          data {
+            size
+            type
+          }
+          block {
+            id
+            timestamp
+            height
+          }
+        }
+        cursor
+      }
+    }
+  }
+`
+
+export async function runStatsTask(context: GatewayContext) {
+  await TaskRunner
+    .from("[stats]", loadStats, context)
+    .runAsyncEvery(STATS_INTERVAL_MS);
+}
+
+async function loadStats(context: GatewayContext) {
+  const {logger, gatewayDb, arweave} = context;
+
+  let results: any[];
+  try {
+    results = await Promise.allSettled([
+      gatewayDb("stats")
+        .select("block_height")
+        .orderBy("block_height", "desc")
+        .limit(1)
+        .first(),
+      arweave.network.getInfo(),
+    ]);
+  } catch (e: any) {
+    logger.error("Error while checking new blocks", e.message);
+    return;
+  }
+
+  const rejections = results.filter((r) => {
+    return r.status === "rejected";
+  });
+
+  if (rejections.length > 0) {
+    logger.error("Error while processing next block", rejections.map((r) => r.message));
+    return;
+  }
+
+  const currentNetworkHeight = results[1].value.height;
+  
+  const lastProcessedBlockHeight = results[0].value?.block_height || 472808;
+  logger.debug(`Last processed block height: ${lastProcessedBlockHeight}`);
+
+  logger.debug("Network info", {
+    currentNetworkHeight,
+    lastProcessedBlockHeight,
+  });
+
+  const heightFrom = parseInt(lastProcessedBlockHeight) - 10;
+  let heightTo = currentNetworkHeight;
+  if (heightTo > heightFrom + 7000) {
+    heightTo = heightFrom + 7000;
+  }
+
+  logger.debug("Loading contracts for blocks", {
+    heightFrom,
+    heightTo,
+  });
+
+  const variables = {
+      tags: [
+        {
+          name: 'App-Name',
+          values: ['SmartWeaveContract']
+        }
+      ],
+      blockFilter: {
+        min: heightFrom,
+        max: heightTo,
+      },
+      first: 100
+    }
+
+  const contracts = await loadPages(context, variables);
+
+  for (let i = 0; i< contracts.length; i++) {
+    try {
+      await gatewayDb("stats")
+        .insert({
+          contract_id: contracts[i].node.id,
+          owner: contracts[i].node.owner.address, 
+          block_height: contracts[i].node.block.height,
+          block_id: contracts[i].node.block.id,
+          timestamp: contracts[i].node.block.timestamp,
+          tags: JSON.stringify(contracts[i].node.tags),
+          data_type: contracts[i].node.data.type,
+          fee: contracts[i].node.fee
+        })
+        .onConflict("contract_id")
+        .merge();
+    } catch (e) {
+      logger.error("Error while loading stats", e);
+
+    }
+  }
+
+}
+
+async function loadPages(
```

Comment:
maybe we should refactor this to some common function, that would be also used in the sync interactions task?